src/lib/HasuraGraphiQL.tsx

```diff
@@ -0,0 +1,563 @@
+        label: "Explorer",
+        title: "Toggle Explorer",
+        onClick: () => setExplorerVisible(!explorerVisible),
+      },
+      {
+        label: "Code Exporter",
+        title: "Toggle Code Exporter",
+        onClick: () => setCodeExporterVisible(!codeExporterVisible),
+      },
+    ];
+    if (isCloud)
+      buttons.push({
+        label: "Cache",
+        title: "Cache the response of this query",
+        onClick: () => setQuery(toggleCacheDirective(query)),
+      });
+    return buttons.map((b) => {
+      return <GraphiQL.Button key={b.label} {...b} />;
+    });
+  };
+
+  React.useEffect(() => {
+    if (loading)
+      fetch(url, {
+        method: "post",
+        headers: transformHeaders(headers),
+        body: JSON.stringify({
+          query: getIntrospectionQuery(),
+        }),
+        credentials: "omit",
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          if (data.errors) setError(data.errors[0].message);
+          else setError(null);
+          setSchema(buildClientSchema(data.data));
+          setLoading(false);
+          setErrorShown(false);
+        })
+        .catch(() => {
+          setErrorShown(true);
+          setSchema(undefined);
+          setLoading(false);
+        });
+  }, [headers, url, loading]);
+
+  return (
+    <div id="hasura-graphiql-wrapper">
+      {errorShown && <ErrorNotification />}
+      <div className="hasura-graphiql-title-holder">
+        {urlCollapsed ? (
+          <span
+            style={{ cursor: "pointer" }}
+            onClick={() => setUrlCollapsed(!urlCollapsed)}
+          >
+            <IconChevronRight />
+          </span>
+        ) : (
+          <span
+            style={{ cursor: "pointer" }}
+            onClick={() => setUrlCollapsed(!urlCollapsed)}
+          >
+            <IconChevronDown />
+          </span>
+        )}
+        <span
+          className="hasura-graphiql-title"
+          onClick={() => setUrlCollapsed(!urlCollapsed)}
+        >
+          GraphQL Endpoint
+        </span>
+        <div
+          style={{
+            display: urlCollapsed ? "none" : "flex",
+            marginTop: "16px",
+            marginBottom: "16px",
+          }}
+        >
+          <div className="hasura-graphiql-endpoint-holder">
+            <button type="button" className="hasura-graphiql-post-button">
+              POST
+            </button>
+            <input
+              type="text"
+              disabled
+              data-testid="endpoint-input"
+              className="hasura-graphiql-endpoint-input"
+              value={urlInput}
+              onChange={(e) => setUrlInput(e.target.value)}
+              onBlur={() => {
+                if (urlInput !== url) {
+                  setUrl(urlInput);
+                  setSchema(undefined);
+                  setLoading(true);
+                }
+              }}
+            />
+          </div>
+          {customToolbar}
+          {isCloud && (
```

Comment:
Why is the relay api only on cloud ?

Reply:
That was decided after a discussion with @wawhal . It was also decided to hide the relay input for now.

Reply:
Oh that's a good catch. It should be `isHasura`, not just `isCloud`.
But for now, we're hiding it because it's just complicating things. We'll include both when we decide to show it.

Reply:
Then, if it's not present, should we remove it ?